src/BuildIt.Forms/BuildIt.Forms.Controls/Platforms/Android/Renderers/PullToRefreshControlRenderer.cs

```diff
@@ -0,0 +1,164 @@
+            {
+                distanceToTriggerRefresh = (int)Math.Min(Element.Height * MaxSwipeDistanceFactor, RefreshTriggerDistance * Resources.DisplayMetrics.Density);
+            }
+        }
+
+        public override bool OnInterceptTouchEvent(MotionEvent e)
+        {
+            var canPullToRefresh = Element.CanPullToRefresh();
+            System.Diagnostics.Debug.WriteLine($"[{nameof(OnInterceptTouchEvent)}] {e.Action} | {nameof(canPullToRefresh)}? {canPullToRefresh}");
+            if (isAnimatingAfterSuccessfulPulToRefresh || !canPullToRefresh)
+            {
+                return true;
+            }
+
+            return base.OnInterceptTouchEvent(e);
+        }
+
+        // https://android.googlesource.com/platform/frameworks/support/+/f25dedc/v4/java/android/support/v4/widget/SwipeRefreshLayout.java
+        public override bool OnTouchEvent(MotionEvent e)
+        {
+            base.OnTouchEvent(e);
+
+            var canPullToRefresh = Element.CanPullToRefresh();
+            System.Diagnostics.Debug.WriteLine($"[{nameof(OnTouchEvent)}] {e.Action} | {nameof(canPullToRefresh)}? {canPullToRefresh}");
+            if (!canPullToRefresh)
+            {
+                return true;
+            }
+
+            var handled = false;
+            switch (e.Action)
+            {
+                case MotionEventActions.Down:
+                    currentPercentage = 0;
+                    downEvent = MotionEvent.Obtain(e);
+                    previousY = downEvent.GetY();
+                    handled = true;
+
+                    break;
+
+                case MotionEventActions.Move:
+                    if (downEvent == null)
+                    {
+                        break;
+                    }
+
+                    var currentPositionY = e.GetY();
+                    var initialPositionY = downEvent.GetY();
+                    var positionYDiff = currentPositionY - initialPositionY;
+                    if (positionYDiff > touchSlop)
+                    {
+                        // User velocity passed min velocity; trigger a refresh
+                        if (positionYDiff > distanceToTriggerRefresh)
+                        {
+                            StartRefresh();
+
+                            handled = true;
+                            break;
+                        }
+
+                        var offsetTop = positionYDiff;
+                        if (previousY > currentPositionY)
+                        {
+                            offsetTop = positionYDiff - touchSlop;
+                        }
+
+                        Element.HandlePullToRefreshDragGesture(offsetTop);
+
+                        previousY = e.GetY();
+                        handled = true;
+                    }
+
+                    break;
+
+                case MotionEventActions.Up:
+                case MotionEventActions.Cancel:
+                    downEvent?.Recycle();
```

Comment:
Is there always a Cancel - are there other scenarios where we need to dispose of the downEvent?